graylog2-server/src/test/java/org/graylog2/contentpacks/facades/ViewFacadeTest.java

```diff
@@ -0,0 +1,291 @@
+import org.graylog.plugins.views.search.SearchRequirements;
+import org.graylog.plugins.views.search.db.SearchDbService;
+import org.graylog.plugins.views.search.elasticsearch.ElasticsearchQueryString;
+import org.graylog.plugins.views.search.filter.OrFilter;
+import org.graylog.plugins.views.search.filter.QueryStringFilter;
+import org.graylog.plugins.views.search.filter.StreamFilter;
+import org.graylog.plugins.views.search.views.DisplayModeSettings;
+import org.graylog.plugins.views.search.views.FormattingSettings;
+import org.graylog.plugins.views.search.views.Titles;
+import org.graylog.plugins.views.search.views.ViewDTO;
+import org.graylog.plugins.views.search.views.ViewRequirements;
+import org.graylog.plugins.views.search.views.ViewService;
+import org.graylog.plugins.views.search.views.widgets.aggregation.AggregationConfigDTO;
+import org.graylog.plugins.views.search.views.widgets.aggregation.AutoIntervalDTO;
+import org.graylog.plugins.views.search.views.widgets.aggregation.TimeHistogramConfigDTO;
+import org.graylog.plugins.views.search.views.widgets.messagelist.MessageListConfigDTO;
+import org.graylog2.bindings.providers.MongoJackObjectMapperProvider;
+import org.graylog2.contentpacks.EntityDescriptorIds;
+import org.graylog2.contentpacks.model.ModelId;
+import org.graylog2.contentpacks.model.ModelTypes;
+import org.graylog2.contentpacks.model.entities.Entity;
+import org.graylog2.contentpacks.model.entities.EntityDescriptor;
+import org.graylog2.contentpacks.model.entities.EntityExcerpt;
+import org.graylog2.contentpacks.model.entities.EntityV1;
+import org.graylog2.contentpacks.model.entities.NativeEntity;
+import org.graylog2.contentpacks.model.entities.SearchEntity;
+import org.graylog2.contentpacks.model.entities.StreamEntity;
+import org.graylog2.contentpacks.model.entities.ViewEntity;
+import org.graylog2.contentpacks.model.entities.ViewStateEntity;
+import org.graylog2.contentpacks.model.entities.references.ValueReference;
+import org.graylog2.database.MongoConnection;
+import org.graylog2.database.MongoConnectionRule;
+import org.graylog2.plugin.cluster.ClusterConfigService;
+import org.graylog2.plugin.indexer.searches.timeranges.KeywordRange;
+import org.graylog2.shared.bindings.providers.ObjectMapperProvider;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+import javax.ws.rs.NotFoundException;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.lordofthejars.nosqlunit.mongodb.InMemoryMongoDb.InMemoryMongoRuleBuilder.newInMemoryMongoDbRule;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ViewFacadeTest {
+
+
+    @ClassRule
+    public static final InMemoryMongoDb IN_MEMORY_MONGO_DB = newInMemoryMongoDbRule().build();
```

Comment:
Because I do not want to build huge `Search` and `ViewDTO` objects. This is real data and found some bugs in my code, which I would not have found if I just have passed some half heartily build DTOs.